substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrNativeEventWriter.java

```diff
@@ -147,10 +146,13 @@ public static void putLong(JfrNativeEventWriterData data, long v) {
     }
 
     @Uninterruptible(reason = "Accesses a native JFR buffer.", callerMustBe = true)
-    public static void putUtf8(JfrNativeEventWriterData data, String string) {
-        int size = UninterruptibleUtils.String.modifiedUtf8Length(string, true);
-        if (ensureSize(data, size)) {
-            UninterruptibleUtils.String.toModifiedUtf8(string, data.getCurrentPos(), data.getEndPos(), true);
+    public static void putUtf8(JfrNativeEventWriterData data, char[] string) {
+        if (ensureSize(data, string.length)) {
```

Comment:
This change seems a bit off. It's now writing a string encoded as a char array, but the method is still named otherwise. Maybe the method name can be changed to `putCharArray`?

Reply:
Changed.